tests/unit-tests/algorithms/dim2/transfer.cpp

```diff
@@ -267,41 +73,103 @@ template <typename Initializer> struct EdgeFunction2D {
-std::vector<std::array<type_real, n_components> >
-compute_transferred_function(const TransferFunction2D &transfer_function,
-                             const EdgeFunction2D &field) {
-  const int n_edges = transfer_function.get_edges();
-  std::vector<std::array<type_real, n_components> > result_field(
-      n_edges, std::array<type_real, n_components>{ 0.0 });
+std::vector<std::array<std::array<type_real, EdgeFunction2D::num_components>,
+                       TransferFunction2D::nquad_intersection> >
+expected_solution(const TransferFunction2D &transfer_function,
+                  const EdgeFunction2D &field) {
+  const int n_edges = TransferFunction2D::num_edges;
+  std::vector<std::array<std::array<type_real, EdgeFunction2D::num_components>,
+                         TransferFunction2D::nquad_intersection> >
+      result_field(
+          n_edges,
+          std::array<std::array<type_real, EdgeFunction2D::num_components>,
+                     TransferFunction2D::nquad_intersection>{ 0.0 });
   for (int i = 0; i < n_edges; ++i) {
-    for (int j = 0; j < n_quad_intersection; ++j) {
-      for (int k = 0; k < n_components; ++k) {
-        for (int l = 0; l < n_quad_element; ++l) {
-          result_field[i][k] += transfer_function(i, j, l) * field(l, j, k);
+    for (int j = 0; j < TransferFunction2D::nquad_intersection; ++j) {
+      for (int k = 0; k < EdgeFunction2D::num_components; ++k) {
+
+        for (int l = 0; l < TransferFunction2D::nquad_edge; ++l) {
+          result_field[i][j][k] += transfer_function(i, l, j) * field(i, l, k);
         }
       }
     }
   }
   return result_field;
 }
 
+/*
+ * Specialization: We are transfering a function f =
+ * AnalyticalFunction::evaluate using an actual quadrature rule.
+ */
+template <typename AnalyticalFunction, typename EdgeQuadraturePoints,
+          typename IntersectionQuadraturePoints>
+std::vector<
+    std::array<std::array<type_real, 1>, IntersectionQuadraturePoints::nquad> >
+expected_solution(
+    const specfem::test::fixture::TransferFunction2D<
+        specfem::test::fixture::TransferFunctionInitializer2D::
+            FromQuadratureRules<EdgeQuadraturePoints,
+                                IntersectionQuadraturePoints> >
+        &transfer_function,
+    const specfem::test::fixture::EdgeFunction2D<
+        specfem::test::fixture::EdgeFunctionInitializer2D::
+            FromAnalyticalFunction<AnalyticalFunction, EdgeQuadraturePoints> >
+        &field) {
+  using TransferFunction2D = specfem::test::fixture::TransferFunction2D<
+      specfem::test::fixture::TransferFunctionInitializer2D::
+          FromQuadratureRules<EdgeQuadraturePoints,
+                              IntersectionQuadraturePoints> >;
+  using EdgeFunction2D = specfem::test::fixture::EdgeFunction2D<
+      specfem::test::fixture::EdgeFunctionInitializer2D::FromAnalyticalFunction<
+          AnalyticalFunction, EdgeQuadraturePoints> >;
```

Comment:
his syntax here could be much shorter and more clear with following change. 

```suggestion
template <typename TransferFunction2d, typename EdgeFunction2D>
std::enable_if_t<is_analytical2d_v<EdgeFunction2D>, std::vector<std::array<std::array<type_real, EdgeFunction2D::num_components>,
                       TransferFunction2D::nquad_intersection> >
expected_solution(
    const TransferFunction2d &transfer_function, const EdgeFunction2d &edge_function)
```

Reply:
Check the syntax. Also add `is_analytical2d_v` helper.

Reply:
That is what I had tried to do initially. The issue was that the function call was ambiguous -- I added a negated std::enable_if_t on the base just now, which solved that.